### PR TITLE
WebCrawler: re-crawl on settings update + garbage collector

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

Re-crawl a website when settings are changed.
This requires adding a garbage collector and factor the website creation UI as a React component to be re-used for website edition

## Deploy Plan
- Lock deployments
- Deploy connectors edge
- run queries to set depth and max pages to all existing web crawler
- run connectors> initdb
- deploy connectors and front

### CODE REVIEW

File: connectors/src/connectors/webcrawler/index.ts [hunk 1]
```diff
@@ -32,6 +37,10 @@ export async function createWebcrawlerConnector(
   dataSourceConfig: DataSourceConfig,
   urlConfig: CreateConnectorUrlRequestBody
 ): Promise<Result<string, Error>> {
+  const depth = urlConfig.depth;
+  if (!isDepthOption(depth)) {
```
Comment (flvndvd) [inline connectors/src/connectors/webcrawler/index.ts:41]: Let's move this logic to the io-ts codec. I added a range codec, like this:

https://github.com/dust-tt/dust/blob/fae4be7564994c04dd77dc019c60aba3e841480a/types/src/front/api_handlers/internal/agent_configuration.ts#L7

File: connectors/src/connectors/webcrawler/index.ts [hunk 2]
```diff
@@ -326,3 +358,45 @@ export async function retrieveWebCrawlerObjectsParents(
...
 }
+
+type GetWebCrawlerConfigurationResBody =
+  WithConnectorsAPIErrorReponse<WebCrawlerConfigurationType>;
+
+async function _getWebcrawlerConfiguration(
+  req: Request<{ connector_id: string }, GetWebCrawlerConfigurationResBody>,
+  res: Response<GetWebCrawlerConfigurationResBody>
+) {
+  const connector = await ConnectorModel.findByPk(req.params.connector_id);
```
Comment (flvndvd) [inline connectors/src/connectors/webcrawler/index.ts:369]: I merged my branch, you'll need to update to:
```suggestion
  const connector = await ConnectorResource.fetchById(req.params.connector_id);
```

File: connectors/src/connectors/webcrawler/temporal/activities.ts [hunk 3]
```diff
@@ -272,3 +277,62 @@ function formatDocumentContent({
     sections: [],
   };
 }
+
+export async function webCrawlerGarbageCollector(
+  connectorId: ModelId,
+  lastSyncStartTsMs: number
+) {
+  const connector = await ConnectorModel.findByPk(connectorId);
```
Comment (flvndvd) [inline connectors/src/connectors/webcrawler/temporal/activities.ts:285]: Same here.

File: connectors/src/lib/models/webcrawler.ts [hunk 4]
```diff
@@ -57,7 +57,7 @@ WebCrawlerConfiguration.init(
     },
     depth: {
       type: DataTypes.INTEGER,
-      allowNull: true,
+      allowNull: false,
```
Comment (flvndvd) [inline connectors/src/lib/models/webcrawler.ts:60]: Will be able to perform this migration in prod? I mean does all the webcrawler configurations have a depth in production?

File: front/components/data_source/WebsiteConfiguration.tsx [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+  const [isSaving, setIsSaving] = useState(false);
+  const [isEdited, setIsEdited] = useState(false);
+  const [isValid, setIsValid] = useState(true);
+
+  const [dataSourceNameError, setDataSourceNameError] = useState("");
+  const [dataSourceUrl, setDataSourceUrl] = useState(
+    webCrawlerConfiguration?.url || ""
+  );
+  const [maxPages, setMaxPages] = useState<number | null>(
+    webCrawlerConfiguration?.maxPageToCrawl || 50
```
Comment (flvndvd) [inline front/components/data_source/WebsiteConfiguration.tsx:57]: If 50 is the maximum, can we use the Macro?

Comment (lasryaric) [reply front/components/data_source/WebsiteConfiguration.tsx:57]: which macro?

File: connectors/src/lib/models/webcrawler.ts (hunk 4 above)
Comment (lasryaric) [reply connectors/src/lib/models/webcrawler.ts:60]: I will manually update them before.

File: front/components/data_source/WebsiteConfiguration.tsx (hunk 5 above)
Comment (lasryaric) [reply front/components/data_source/WebsiteConfiguration.tsx:57]: Nevermind, just got it. I fixed the code to use it at the parsing site.

**Comment by lasryaric** [comment]
@flvndvd merging since I addressed all your comments and need to ship that 😬

**Pull request merged**